samples/file-encryptor/enclave/encryptor.h

```diff
@@ -12,8 +12,6 @@
 
 using namespace std;
 
-#define IV_SIZE 16
```

Comment:
I would actually keep this as a separate definition for clarity: fundamentally, there's no reason why the salt size needs to be the same as the IV size, it's just a choice that this particular implementation makes to reuse the salt as the IV.

A little bit of background FYI:
* A _salt_ usually refers to a public, random value used in hashing to mitigate hash table/[rainbow table](https://en.wikipedia.org/wiki/Rainbow_table) attacks on the hashed output. It can be of arbitrary length, and usually chosen to be sufficiently long to make it expensive to generate a comprehensive rainbow table. 
* An _IV (initialization vector)_ is also a public value, used in encryption to produce different ciphertext give the same plaintext and key. As such, it is expected not to be reused across encryptions on the same key, and is sometimes chosen randomly for this reason. Its length is usually a fixed value based on the encryption algorithm and block chaining mode. 

In this case, the implementation is really doing this:

```c
#define IV_SIZE 16 // determined by AES256-CBC
#define SALT_SIZE_IN_BYTES IV_SIZE
```

Because this is a sample, I would choose to make that clear in the code as opposed to defining the IV buffers in terms of salt size.

Also just as FYI, while it's okay in this implementation to use the same value for salt and IV, usually it's clearer to keep them separate. A common construction of this when we don't want to store separate salt and IV values is to:
* take a salt value of chosen length that can be independent of IV length
* use the salt derive a longer value than needed when using the PBKDF (e.g. 512-bits instead of 256-bits)
* use half the bits as the key as originally intended (password key in this case) and the extra bits as the IV.

Reply:
Moved IV_SIZE to shared.h. Using IV_SIZE in encyptor.cpp, in place of SALT_SIZE_IN_BYTES.